langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/gemini/GeminiAiChatLanguageModel.java

```diff
@@ -0,0 +1,254 @@
+package dev.langchain4j.model.gemini;
+
+import dev.langchain4j.Experimental;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.ResponseFormat;
+import dev.langchain4j.model.chat.request.ResponseFormatType;
+import dev.langchain4j.model.chat.request.json.JsonSchema;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.output.FinishReason;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+import lombok.Builder;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.gemini.PartsAndContentsMapper.fromMessageToGContent;
+import static dev.langchain4j.model.gemini.FinishReasonMapper.fromGFinishReasonToFinishReason;
+import static dev.langchain4j.model.gemini.PartsAndContentsMapper.fromGPartsToAiMessage;
+import static dev.langchain4j.model.gemini.SchemaMapper.fromJsonSchemaToGSchema;
+import static java.util.Collections.emptyList;
+
+@Experimental
+public class GeminiAiChatLanguageModel implements ChatLanguageModel {
+    public static final String GEMINI_AI_ENDPOINT = "https://generativelanguage.googleapis.com/v1beta/";
+    private final String apiKey;
+    private final String modelName;
+
+    private final Double temperature;
+    private final Integer topK;
+    private final Double topP;
+    private final Integer maxOutputTokens;
+    private final List<String> stopSequences;
+
+    private final Integer candidateCount;
+
+    private final String responseMimeType;
+    private final ResponseFormat responseFormat;
+
+    private final GeminiFunctionCallingConfig toolConfig;
+
+    private final Boolean logRequestsAndResponses;
```

Comment:
I'm relying on Retrofit and OkHttp's own logging, and it doesn't offer the choice of logging either the input or the output, so I went with logging both.
I guess it could be done, but I'd have to write my own interceptor, which I wanted to avoid as it's not necessarily trivial.